Connect Google Cloud SQL to Go API

swoldemi opened the pull request:
This PR closes #1 and closes #3

swoldemi committed:
Ignore env and Cloud SQL service account

swoldemi committed:
Add MustInit function in database package

  - Functions prefixed with `Must` are typically allowed to `panic` outside of `func main`

  - Added Routes struct to encapsulate how API Handlers are defined (main.go#L47,54-55)

ynigoreyes commented on pkg/database/db.go:
```diff
@@ -1,15 +1,40 @@
+	}
+	dataSourceName := fmt.Sprintf("%s:%s@cloudsql(%s)/%s?charset=utf8&parseTime=True&loc=Local", username, password, connectionName, databaseName)
+
+	db, err := gorm.Open("mysql", dataSourceName)
```
Does this mean that the development db is also Cloud SQL? Should we not use a local DB?

swoldemi replied:
Yes, but can create a dev db in the cloud instance. The instance houses several databases. We would just need to change an environment variable. `main` would be prod, and `dev` could be used for development.

ynigoreyes reviewed (changes requested):
Can you please update the README to make sure we keep track of what we did to get Cloud SQL running?

ynigoreyes commented on .gitignore:
```diff
@@ -1,4 +1,6 @@
\ No newline at end of file
+**gin-bin*
+.env
+resume-engine-cloud-sql.json
```
Can you send me this? Or how do I get it on my own?

swoldemi replied:
Sent. I think you can redownload it from IAM > Service Accounts too.

ynigoreyes commented on pkg/database/db.go:
```diff
@@ -1,15 +1,40 @@
-func Init() (*sql.DB, error) {
-	db, err := sql.Open("mysql", "root@cloudsql(project-id:instance-name)/dbname")
+// MustInit initializes an instance of the database
+func MustInit() *gorm.DB {
```
Yeah i remember Josh saying that func Init shouldn't return anything and I was about to ask you lol

swoldemi replied:
Changed in 64452d2c75d13566f91b0608d34ae9873426a337

ynigoreyes commented:
Also, since you are doing the Environments too, take into account that you can do development deployments using `app.yaml` and you can declare production environment variables in app.yaml. If we were able to ignore changes to the app.yaml and then populate it with production environment variables, that would be ideal. I tried this right before you started working on it haha

swoldemi committed:
Rename MustInit to Create

swoldemi committed:
Update README.md with instructions to connect to Cloud SQL

Resolves https://github.com/ynigoreyes/resume-engine/pull/9#pullrequestreview-211500043
